### PR TITLE
multi asset enum

### PR DESCRIPTION
**Pull request by gilescope**
Allow swapping with native. (Ideally rather than this enum we could use just multilocation and a couple of convert functions)

### CODE REVIEW

File: frame/dex/src/lib.rs [hunk 1]
```diff
@@ -413,9 +424,22 @@ pub mod pallet {
...
+					},
+				}
+				match asset2 {
+					MultiAssetId::Native => {
+						T::Currency::transfer(&pallet_account, &withdraw_to, amount2, false)?;
+					},
+					MultiAssetId::Asset(asset2) => {
+						T::Assets::transfer(asset2, &pallet_account, &withdraw_to, amount2, false)?;
+					},
+				}
```
Comment (gilescope) [inline frame/dex/src/lib.rs]: Could pull these out into a transfer function that takes a MultiAsset.

**Comment by gilescope** [comment]
While this passes all the tests, this currently relies on AssetBalance and Currency Balance being the same type. (This is the case for statemine/t fyi)

File: frame/dex/src/lib.rs [hunk 2]
```diff
@@ -307,8 +311,22 @@ pub mod pallet {
...
-				T::Assets::transfer(asset1, &sender, &pallet_account, amount1, false)?;
-				T::Assets::transfer(asset2, &sender, &pallet_account, amount2, false)?;
+				match asset1 {
+					MultiAssetId::Native => {
+						T::Currency::transfer(&sender, &pallet_account, amount1, false)?;
+					},
+					MultiAssetId::Asset(asset1) => {
+						T::Assets::transfer(asset1, &sender, &pallet_account, amount1, false)?;
+					},
+				}
```
Comment (jsidorenko) [inline frame/dex/src/lib.rs]: I would create a helper transfer method for that

File: frame/dex/src/lib.rs (hunk 1 above)
Comment (jsidorenko) [reply frame/dex/src/lib.rs]: yeah, I think it's a good idea

**Review by jsidorenko — APPROVED** [review]

**Pull request merged**